### ISSUE TITLE
Alter Code To Short Using Different Exchange

### ISSUE THREAD

**Issue by taiez** [butor/blackbird#344]
Hi Everybody,

I found this arbitrage bot a few weeks ago and was messing around with installing and running (in demo mode) it on an Amazon EC2 instance. I was successful, and the demo mode was showing several arbitrage opportunities. When I went to create a Bitfinex account to start trading with real money I was unable to. Apparently, they aren't taking new members until Jan 15th, if you believe that. This leads me to my question:

Is there an easy way to alter the code to use a different trading exchange as the shorting exchange? 

I'll admit that I haven't looked at the code yet. It has been a while since I've seen a C++ code. However, I would like to get to know the guts of the code, and would enjoy altering it a bit. If it is not an easy endeavour though, I might just try to wait out Bitfinex...

Thanks in advance everyone! This is an amazing bot with what seems like a passionate community. Hopefully, I can get in the game too!

**Comment by butor** [comment]
Hi,

[CEX.IO](https://cex.io/) seems to be a good candidate: They are based in the UK and here is their [REST API documentation](https://cex.io/rest-api). Short selling is possible.

**Comment by Mgla96** [comment]
Hello,

This is some great work! Would Poloniex margin trading be able to be supported? CEX.IO margin trading doesn't seem to be working for U.S customers.

**Comment by cricc78** [comment]
@butor I'm willing to work on adding CEX.IO to Blackbird (currently I'm ~25% complete), however, I need an API key / secret to finish and fully test. CEX.IO has temporarily disabled new sign-ups to their exchange so I can't get the key / secret. 

Does anyone know a way around this?

**Comment by butor** [comment]
I don't see any possible workaround, and I don't have any account on CEX.IO that you could temporary use.

Hopefully they will reopen their new user registration soon ([more info here](https://blog.cex.io/news/registration-on-hold-16870)).

**Comment by cricc78** [comment]
Yep, I'll just create an account once they're back up.

**Comment by cricc78** [comment]
BTW - it appears as if CEX.IO BTC/USD is generally priced much higher than on the other exchanges - at least that was my observation over the past couple of days.  Not sure if this will last or if it will normalize over time, but this exchange does appear to be a good one considering it allows short positions.

Whether or not they will allow us US-based folks to trade on margin is another question...

**Comment by blundell** [comment]
related to #341

**Comment by blundell** [comment]
related to #324

**Comment by blundell** [comment]
related to #307

**Comment by blundell** [comment]
related to #302

**Comment by gweit** [comment]
Hi, is short selling on okcoin already implemented?

**Comment by andygoossens** [comment]
No. But don't get your hopes up too high as okcoin lacks liquidity.

**Comment by mattsicotte** [comment]
Hey Gang, I have had an active CEX.io account for a while(use to be GHash.io), I would love to have it implemented in this bot, how may we do that?

**Comment by JinjaNinja22** [comment]
Any update on progress of adding CEX.io ?

**Comment by cricc78** [comment]
I’ve got a good chunk of the code done; however, I still need an API key to test everything out. I’m sure code refinement will be necessary once I can test. 

Last I checked, CEX.IO is still not accepting new accounts so I haven’t been able to get my own API key.

**Comment by JinjaNinja22** [comment]
oh right i see. I do have a CEX.io account that i dont have any money on currently.

**Comment by cricc78** [comment]
@JinjaNinja22 OK, let me know the best way to contact you and we can discuss.

**Comment by JinjaNinja22** [comment]
@cricc78 sorry for the delay, fibre was down. Sucks that there aren't DMs on here anymore, do you have twitter? Can DM there to discuss.

**Comment by cricc78** [comment]
Yep, it's a pain. Sure, you can DM here: [https://twitter.com/chrisct]

**Comment by JinjaNinja22** [comment]
@cricc78 sorry for the delay, i followed you on twitter [https://twitter.com/merrick_bengis]. Follow back when you get a chance so we can DM.

**Comment by taiez** [comment]
Any lick here?

**Comment by cricc78** [comment]
Making progress. Probably another couple of weeks, assuming I don’t get stuck anywhere. But then I’ll ask you guys for help!

Currently testing / optimizing the buy / sell orders.

**Comment by flocoder** [comment]
Hi Chris,
I am just wondering how is going with your setting up Blackbird? 
Have you done any trade?
Is it worth it?
Thanks

**Comment by cricc78** [comment]
@flocoder No, I have not executed any real money trades beyond what I did while testing the Cex.io API. The main problem is that as a US resident, not many exchanges allow me to sell short. The ones that do allow shorts for US residents tend to price bitcoin lower than other exchanges, based on my recent observations. Blackbird works in part by initiating a short on an exchange that prices bitcoin higher than another...but if I can't initiate shorts on those exchanges, it leaves me with no way to exploit the opportunity.

I'm still working on finding an opportunity here, but if you're not from the US, I would give it a go and see what results you can obtain. You may be better positioned than I because you can probably initiate shorts on the exchanges that tend to price bitcoin higher than others.

**Comment by brettham70** [comment]
Any closer to being able to test with short selling with CEX.io ? any help needed with testing or known bugs ?

**Comment by fdoumet** [comment]
As was noted in #392, the minimum to short on CEX is 0.05 BTC. I haven't heard of anyone who has tested with that amount yet. Testing with any smaller amount will cause the CEX API to return an error and Blackbird to crash.

**Comment by brettham70** [comment]
@fdoumet - I tested it yesterday with 0.05 or larger amount in account (USD1000). Order still received the same error. Suggest it's not calling the correct API and assume if i had put BTC in account it would have maybe sold it.Problem i see with CEX is that the ARB % is always there between 1.5% and 3% currently meaning positions will never close. Anyone else with a good ARB idea let me know :).

**Comment by redpanda321** [comment]
I think  CEX.IO  short must use this interface below   not this  (/place_order/BTC/USD/)

Open position
POST https://cex.io/api/open_position/{symbol1}/{symbol2}
